### ISSUE TITLE
json field names cannot be substrings of other field names??

### ISSUE THREAD

**Issue by jlmalone** [xamoom/Morpheus#9]
For the life of me I have tried to set up my model objects correctly, but it seems since my json response has fields 

```
@SerializedName("starts-ends-at-timezone")
String startsEndsAtTimezone;
```

and 

```
@SerializedName("ends-at")
String endTime;
```

Any object that I create that includes the latter field, ends up creating through morpheus objects full of null values. I believe it is some quirk where since "ends-at" is an inner substring of "starts-ends-at-timezone" the library is trying to parse in a bad way and failing out with a caught exception that hides the error.

My short term fix will be to ask my API team to change the strings, but this issue really needs to be fixed in the library.

**Comment by xamoom-raphael** [comment]
Hey thanks. I will be working on later today, the i will check this problem.

**Comment by jlmalone** [comment]
It may be related or it may be a separate issue. 

My json spec looks like:

"attributes": {
    ...
    "actions": [
        {
            "title": "Get that T-Shirt",
            "url": "http://shop.lululemon.com/product/2345"
        },  
        {
            "title": "Get Tickets",
            "url": "http://eventbrite.com/get-tickets/125341"
        }
    ]
}

So far I have just tested with 

"actions" : null

but including 

@SerializedName("actions")
ArrayList<Action> actions;

also causes the object to fail to parse correctly. Is this a problem of having an array or complex object inside the attributes rather than as part of a relationship or should the library be able to handle this scenario properly?

**Comment by xamoom-raphael** [comment]
I tried yesterday to reproduce the first problem with the names, but I was not able to reproduce it.
Can you send me a complete json, what I can use for my tests?

**Comment by jlmalone** [comment]
Thank you! I sent you an email with a lot of details.

**Comment by jlmalone** [comment]
So I went through with the debugger. I do not think the names have anything to do with my problem. I believe that the inclusion of nulls in my attributes is what is causing my problem. Specifically, the attribute mapper attempts to catch JSONExceptions but

in at.rags.morpheus.AttributeMapper we have

```
 /**
   * Will map the attributes of the JSONAPI attribute object.
   * JSONArrays will get mapped as {@literal List<Object>}.
   * JSONObject will get mapped as {@literal ArrayMap<String, Object>}.
   * Everything else will get mapped without changes.
   *
   * @param jsonApiResource Object extended with {@link Resource} that will get the field set.
   * @param attributesJsonObject {@link JSONObject} with json:api attributes object
   * @param field Field that will be set.
   * @param jsonFieldName Name of the json-field in attributesJsonObject to get data from.
   */
  public void mapAttributeToObject(Resource jsonApiResource, JSONObject attributesJsonObject, Field field, String jsonFieldName) {
    try {
      if (attributesJsonObject.get(jsonFieldName).getClass() == JSONArray.class) {
        List<Object> list = createListFromJSONArray(attributesJsonObject.getJSONArray(jsonFieldName), field);
        mDeserializer.setField(jsonApiResource, field.getName(), list);
      } else if (attributesJsonObject.get(jsonFieldName).getClass() == JSONObject.class) {
        Gson gson = new Gson();
        Object obj = gson.fromJson(attributesJsonObject.get(jsonFieldName).toString(), field.getType());
        mDeserializer.setField(jsonApiResource, field.getName(), obj);
      } else {
        mDeserializer.setField(jsonApiResource, field.getName(), attributesJsonObject.get(jsonFieldName));
      }
    } catch (JSONException e) {
      Logger.debug("JSON attributes does not contain " + jsonFieldName);
    }
  }
```

The method is meant to catch JSONException. However gson.fromJson throws a JsonSyntaxException extends JsonParseException extends RuntimeException 

also something is throwing an IllegalArgumentException which must also be caught

In other words you are not catching the right exception so any null values end up jumping out to Factory.newObjectFromJSONObject(...) which catches generic exceptions.

**Comment by jlmalone** [comment]
If you give me priviledges I can open a PR

**Comment by xamoom-raphael** [comment]
Hey you can fork this repo and then make a PR without any need of privileges.
[Pull Request from Fork Help Link](https://help.github.com/articles/creating-a-pull-request-from-a-fork/)

**Comment by jlmalone** [comment]
yeah, I just figured I would rather contribute a fix rather than confuse the world with an extra fork if it makes sense...

I think if I wanted to make a lot of major changes I would go that route, but up to you...

**Comment by xamoom-raphael** [comment]
You can, but do it so:
- Fork it
- Create a branch and make the changes
- Make a pull request here

I will look over it and then merge it into Morpheus.
